Checking terminology-server for official translations

Frontman50 opened the pull request:
- added TerminologyDesignationResolver.py for checking translations on server
- added (--terminology_server_config) for terminology_resolver
- added (--use_available_translations) for toggling server translations checking

Frontman50 committed:
first draft, but basically ready
- added TerminologyDesignationResolver.py for checking translations on server
- added (--terminology_server_config) for terminology_resolver
- added (--use_available_translations) for toggling server translations checking

juliangruendner commented:
@Frontman50  - please use more professional naming of the PR

juliangruendner commented on translate.py:
```diff
@@ -21,6 +23,9 @@ def configure_args_parser():
     arg_parser.add_argument('--dry_run', action='store_true', help='Do not translate, only count number of characters that would be translated')
 
+    arg_parser.add_argument("--terminology_server_config", type=str, help="The terminology server config", default="terminology_server_config.json")
+    arg_parser.add_argument("--use_available_translations", action='store_true', help='Use available translations from specified terminology server')
```
@Frontman50  - created - but appears to be unused

juliangruendner commented on terminology_server_config.json:
```diff
@@ -0,0 +1,30 @@
+{
```
is ais the version parameter not missing here for sct ?
@Frontman50

juliangruendner commented on translate.py:
```diff
@@ -47,11 +52,19 @@ def configure_logging(log_level_arg):
     nr_of_translated_files = 0
     characters_translated = 0
 
+
```
@Frontman50  unnecessary space

juliangruendner commented on Translator.py:
```diff
@@ -69,47 +91,61 @@ def translate(self, value_set_url, source_lang, dry_run , batch_size=5):
+            for concept_code, concept_content in translated_concepts.items():
+                i += 1
+                if not concept_content.get(language) or concept_content.get(language) == "":
+                    batch_for_ai[concept_code] = concept_content
```
@Frontman50  strange naming - change to
batch_to_translate

juliangruendner commented on translate.py:
```diff
@@ -47,11 +52,19 @@ def configure_logging(log_level_arg):
-    translator = Translator(args.deepl_api_key, session, args.terminology_server, ["de", "en"])
+    translator = Translator(args.deepl_api_key, session, args.terminology_server, target_langs=["de", "en"],terminology_server_config=args.terminology_server_config)
+
+    if translator.terminologyResolver:
```
@Frontman50  - why the if?

juliangruendner commented on Translator.py:
```diff
@@ -69,47 +91,61 @@ def translate(self, value_set_url, source_lang, dry_run , batch_size=5):
+                        text.append(content.get('display'))
+                        char_count = char_count + len(content.get('display'))
+
+                    if not dry_run:
```
@Frontman50  i think that for a dry_run - the translation should be carried over from the original and not as is currently being implemented set to null

juliangruendner commented on translate.py:
```diff
@@ -47,11 +52,19 @@ def configure_logging(log_level_arg):
+    translator = Translator(args.deepl_api_key, session, args.terminology_server, target_langs=["de", "en"],terminology_server_config=args.terminology_server_config)
+
+    if translator.terminologyResolver:
+        translator.terminologyResolver.load_designations(update_translation_supplements=True)
```
@Frontman50 update_translation_supplements should be configurable via a script param

juliangruendner reviewed (commented):
@Frontman50, @paulolaup  - there should be at least one test included in the repo, which checks the translation and includes a value set which contains at least one concept for which a translation exists in the original

juliangruendner committed:
update value sets json and gitignore

Frontman50 committed:
fixing bugs,
- adding version parameter:

{
   "name": "version",
   "valueString": "http://snomed.info/sct/11000274103"
}
example of working codes: 44532001, 38577009 with this sct version

- fixing the designation_extractor

Frontman50 committed:
refactoring

Frontman50 committed:
refactoring

Frontman50 committed:
refactoring, fixing mentioned issues
